common/archiver/s3store/mocks/S3API_mock.go

```diff
@@ -0,0 +1,25 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package mocks
+
+//go:generate mockery --dir ../../../../vendor/github.com/aws/aws-sdk-go/service/s3/s3iface --name S3API --output .
```

Comment:
seems kinda unrelated?  also may be better to stick to a single mock generator tool, unless there some major benefit of mockery vs mockgen. 

I can never remember which does what, maybe mockery would be worth adopting?  just doesn't seem related to gowrap / should probably be a separate thing.

Reply:
I'll put it in a separate diff. Unfortunately, we use mockery in multiple places in the code.

Reply:
It's merged in https://github.com/uber/cadence/pull/4887